recipes/awscli2/meta.yaml

```diff
@@ -0,0 +1,143 @@
+{% set name = "awscli" %}
+{% set version = "2.13.10" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+# v2 is not on PyPI https://github.com/aws/aws-cli/issues/6785
+source:
+  url: https://github.com/aws/aws-cli/archive/{{ version }}.tar.gz
+  sha256: 523646f7b86b4beef74c0f0e9a2a178ea1d40a21b7da0e6852db17d76044b07c
+
+build:
+  number: 0
+  # Build uses pep517 with a custom backend
+  # console_scripts entrypoints aren't used, so this can't be noarch
+  script: {{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv  # [not unix]
+  script: {{ PYTHON }} -m pip install . --no-deps --ignore-installed -vv && mkdir -p $PREFIX/share/bash-completions/completions && echo complete -C aws_completer aws >$PREFIX/share/bash-completions/completions/aws  # [unix]
+
+requirements:
+  build:
+    - python                                 # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}     # [build_platform != target_platform]
+    # The autocomplete index step is run on the build host
+    - colorama >=0.2.5,<0.4.7                # [build_platform != target_platform]
```

Comment:
we can make this change once it's a feedstock, but I might DRY this a bit by putting each requirement line in a jinja variable. not sure if grayskull would pick up the repeats, or be able to update them if moved to jinja variables though

Reply:
No, grayskull would only update the dependencies in the `run` section. For maintainability, I think, we should up the autocompletion into its own feedstock. Then we should be able to have both maintained via the bot.

Reply:
That sounds like a good idea in concept, and I see @manics put a thumbs up on it. I guess it would use the awscli v2 release tags like this recipe, and depend on the matching version of awscli v2? - so it'd resemble the tied dependency that boto3 and awscli v1 have on botocore right now. Or are there some feedstocks that release multiple packages from the same repo and release tagging?

We can continue this discussion inside the new feedstock once it's created, I don't consider any refactoring of the autocompletion details as blocking

Reply:
I have pushed a second recipe to this PR to work together with the main `awscli2` recipe.

Reply:
That is a much cleaner refactoring, I like that. @manics any thoughts on it? Is it much of a usability issue in your opinion to need a separate package install for autocompletion?

Automerge won't be too useful on the autocomplete package, the PR's will usually get opened at the same time and the autocomplete one will fail until the corresponding awscli2 release goes through and gets published, but I'm used to that dependency with botocore/boto3/awscli1

Reply:
I'm probably not best placed to decide on the usability of installing autocompletion, I only found out about it when I opened the original v2 PR!

However we could swap the order of the packages around- make `awscli2-core` (or some other name such as `awscli2-base`) contain the AWS CLI, and `awscli2` be the autocompletion package that depends on `awscli2-core`. This means `conda install awscli2` would work as expected.

Reply:
> I might DRY this a bit by putting each repeated requirement line in a jinja variable. not sure if grayskull would pick up the repeats, or be able to update them if moved to jinja variables though

I went and did this little refactoring in https://github.com/conda-forge/awscli2-feedstock/commit/6757b3ab22904ea0f08a48552449ee7d8ed8cbde
Unfortunately grayskull doesn't seem to be understanding these dependencies well enough to update them either before or after that change.

Reply:
Probably, we want to enable grayskull but not automerge. We will see in the diff whether grayskull made any changes and we can then apply them to `host` / `build` manually.

Reply:
It turns out grayskull is erroring because the package isn't on pypi. Need to figure out if there's a way to configure the autotick bot's usage of grayskull to pass parameters so it tries the github codepaths.